api/src/main/java/org/pmiops/workbench/google/CloudResourceManagerServiceImpl.java

```diff
@@ -0,0 +1,71 @@
+package org.pmiops.workbench.google;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.services.cloudresourcemanager.CloudResourceManager;
+import com.google.api.services.cloudresourcemanager.model.Project;
+import com.google.api.services.cloudresourcemanager.CloudResourceManagerScopes;
+import org.pmiops.workbench.auth.Constants;
+import org.pmiops.workbench.auth.ServiceAccounts;
+import org.pmiops.workbench.db.model.User;
+import org.pmiops.workbench.exceptions.ExceptionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Provider;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.google.api.client.googleapis.util.Utils.getDefaultJsonFactory;
+
+@Service
+public class CloudResourceManagerServiceImpl implements CloudResourceManagerService {
+  private static final String APPLICATION_NAME = "All of Us Researcher Workbench";
+
+  private static final List<String> SCOPES = Arrays.asList(
+      CloudResourceManagerScopes.CLOUD_PLATFORM_READ_ONLY
+  );
+
+  private final Provider<GoogleCredential> cloudResourceManagerAdminCredsProvider;
+  private final HttpTransport httpTransport;
+  private final GoogleRetryHandler retryHandler;
+  private final ServiceAccounts serviceAccounts;
+
+  @Autowired
+  public CloudResourceManagerServiceImpl(
```

Comment:
For testing implementations of Google services, we've tended to skip unit tests and only have integration-level tests. It may make sense to do the same here. See the integration/j/o/p/workbench/ test files for examples.

A couple options for testing this scenario:

1. Assume there is a test user "cloud-resource-manager-test@fake-research-aou.org" with a single billing project, and make the ListAllProjects call on behalf of this user. Ensure the list contains 1 project w/ the expected project name. (The trouble here is that if we ever inadvertently clear out users from our test environment, this test would start to break.)
2. Same as the above, but do as much setup & teardown within the test as you can (instead of assuming the existence of a project / user). This may require way more effort, to the point of not being worthwhile.

@aweng98 I'd love to hear if you have any other suggestions for integration testing here...

Reply:
I agree. thumb up for adding integration tests. makes perfect sense to me.
The code relies on the contract of a external service (Google Cloud Resource API), integration test that exercising Google API service calls increases our confidence that Google services are working as expected and is used correctly.

Reply:
`GoogleCredential` provides a `getApplicationDefault`. maybe that instead providing our own test user like "cloud-resource-manager-test@fake-research-aou.org".